punjab/session.py

```diff
@@ -672,9 +672,9 @@ def checkExpired(self):
             reactor.callLater(wait, self.checkExpired)
 
     def _cacheData(self, rid, data):
-        if len(self.cache_data.keys()) >= 3:
+        if len(list(self.cache_data.keys())) >= 3:
```

Comment:
Can we change this code to the following?

```
cache_keys = list(self.cache_data.keys())
if len(cache_keys) >= 3:
    keys.sort()
    del self.cache_data[keys[0]]
```

Reply:
Of course.